components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/handlers/SCIMClaimOperationEventHandler.java

```diff
@@ -104,4 +119,37 @@ public String getName() {
 
         return "SCIMClaimOperationEventHandler";
     }
+
+    private void handleSCIMExternalClaimAddEvent(Event event) {
+
+        if (!IdentityEventConstants.Event.PRE_ADD_EXTERNAL_CLAIM.equals(event.getEventName())) {
+            return;
+        }
+
+        String claimDialectUri = (String) event.getEventProperties()
+                .get(IdentityEventConstants.EventProperty.CLAIM_DIALECT_URI);
+        String externalClaimUri = (String) event.getEventProperties()
+                .get(IdentityEventConstants.EventProperty.EXTERNAL_CLAIM_URI);
+        Set<String> scimClaimDialects = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(
```

Comment:
We may be able to define this as final static variable.

Reply:
Fixed with 5af70228808cebfd3bc272179fce9a76e586830f